segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/SlowStorageFactory.java

```diff
@@ -51,8 +53,17 @@ public class SlowStorageFactory implements SimpleStorageFactory {
     @Override
     public Storage createStorageAdapter(int containerId, ChunkMetadataStore metadataStore) {
         if (inner instanceof SimpleStorageFactory) {
-            val innerStorage = ((SimpleStorageFactory) inner).createStorageAdapter(containerId, metadataStore);
-            return new SlowStorage(innerStorage, executor, config);
+            if (config.isSlowModeInjectChunkStorageOnly()) {
```

Comment:
is the config.isSlowModeInjectChunkStorageOnly redundant, as its checked in createChunkStorage as well?

Reply:
SLTS has two sides
- Incoming calls from segment storage- `Storage` is the interface meant for the callers of SLTS i.e rest of the segment store.
- Outgoing calls to LTS - `ChunkStorage` is the interface individual LTS adapters implement.

In between those two end points there is 10K+ lines of SLTS code that converts call to Storage into appropriate interactions with LTS, manages metadata and deals with all sorts of failure scenarios etc.

So we could inject delays in two places
1. SlowChunkStorage  - add delays to actual calls going out to LTS - This delays only the single LTS calls.
2. SlowStorage - adds delays to all incoming Storage calls. This delays entire storage call.

So, No - config.isSlowModeInjectChunkStorageOnly is not redundant. It is used to decide whether we narrowly slow down only the actual LTS calls or slow down entire storage call.